### DISCUSSION TITLE
Provenance and super rares?

### DISCUSSION THREAD

**Discussion by ultraviolet007rainbow** [nftchef/art-engine#91] [Help + Q&A] [answered]
Hi,

I generated Provenance hashes for all my images.
If I wanted to insert super-rares using the utility, they won't have provenance hashes.
Is there a way to generate provenance hashes for 1/1 super rares other than running them through the generator I guess 1 by 1?

Thanks.

**Comment by nftchef** [comment, marked as answer]
The utility script for inserting rares, `utils/replace.js` will generate an `imageHash` when it replaces the image, then, you are able to run the provenance util to get the concatenated string and provenance hash with the rares included.

**Comment by ultraviolet007rainbow** [comment]
Wow you think of everything you are awesome!
Thank you.
